Fix unittests and clean transactions in sqlite database

tawmaz opened the pull request:
- Clean expired transactions from SQLite Payload and Transactions db

- Fix error in SQLite query for orphaned transactions

- Enable mempool_tests

- Add verifydb function to validate sqlite transactions are present on the blockchain

- Additional debugging and check mempool for existance of transactions

- Re-arrange locks in LoadMemPool to be consistent with lock usage through rest of code

- Refactor CleanSQLite function and print error if transaction to delete is present in blocks or mempool

- Verify DB deletes orphan transactions and add removal reason to CleanSQLite logging

- Use GetTransaction call to verify TX is valid

- Fix segmentation fault caused by accessing mempool TX which has been removed

- Enable unittests txvalidation_tests, txvalidationcache_tests, and validation_block_tests

- Remove unused parameter from ProcessNewBlock

- Add test pocketnet_block_tests

- pocketnet_block_tests: test InvalidateBlock and adding block to chain

- Add transactions to block from mempool

- Fix execution of unittests and disable Segwit on regtest for now

- Fix pocketnet_block_tests error by setting up mocktime for tests

- Intermittent test failures were occuring because blocks were generated with same timestamp and same hash value due to one second time stamp granularity.  Now timestamp is incremented for each block.

- Enable wallet tests.

tawmaz committed:
Fix orphan transactions in SQLite database and add unitttest

- Clean expired transactions from SQLite Payload and Transactions db

- Fix error in SQLite query for orphaned transactions

- Enable mempool_tests

- Add verifydb function to validate sqlite transactions are present on the blockchain

- Additional debugging and check mempool for existance of transactions

- Re-arrange locks in LoadMemPool to be consistent with lock usage through rest of code

- Refactor CleanSQLite function and print error if transaction to delete is present in blocks or mempool

- Verify DB deletes orphan transactions and add removal reason to CleanSQLite logging

- Use GetTransaction call to verify TX is valid

- Fix segmentation fault caused by accessing mempool TX which has been removed

- Enable unittests txvalidation_tests, txvalidationcache_tests, and validation_block_tests

- Remove unused parameter from ProcessNewBlock

- Add test pocketnet_block_tests

- pocketnet_block_tests: test InvalidateBlock and adding block to chain

- Add transactions to block from mempool

- Fix execution of unittests and disable Segwit on regtest for now

- Fix pocketnet_block_tests error by setting up mocktime for tests

- Intermittent test failures were occuring because blocks were generated with same timestamp and same hash value due to one second time stamp granularity.  Now timestamp is incremented for each block.

- Enable wallet tests.

tawmaz committed:
New user account creation in pocketnet_social_tests and code formating

tawmaz committed:
Fix nested sqlite transaction in CleanExpiredTransactions

andyoknen commented on src/pocketdb/repositories/TransactionRepository.cpp:
```diff
@@ -459,6 +459,29 @@ namespace PocketDb
+
+            // Get list of expired transaction hashes which have not been added to a block
+            auto stmt = SetupSqlStatement(R"sql(
+                select Hash, Time from Transactions t where t.Hash isnull and t.Height isnull and t.Time < ?
```
@tawmaz  I think there's a typo here, meaning `t.BlockHash isnull`?

tawmaz replied:
This code needs some more work.  I will remove it from this patch an resubmit.

andyoknen commented on src/pocketdb/repositories/TransactionRepository.cpp:
```diff
@@ -459,6 +459,29 @@ namespace PocketDb
+            )sql");
+            TryBindStatementInt64(stmt, 1, time);
+            while (sqlite3_step(*stmt) == SQLITE_ROW)
+                hashes.push_back( std::string(reinterpret_cast<const char*>(sqlite3_column_text(*stmt, 0))) );
```
@tawmaz It might be better to use `if (auto[ok, value] = TryGetColumnText(*stmt, 0); ok)` ?

tawmaz replied:
Removed from latest commits

lostystyg commented on src/pocketdb/repositories/ChainRepository.h:
```diff
@@ -45,6 +45,10 @@ namespace PocketDb
         // Erase all calculated data great or equals block
         bool Rollback(int height);
 
+        // Get hash value of Transaction of a given row in the database, used to validate
```
A bit frustrating description, it gets hashes of all transactions, not of a given row.

tawmaz replied:
This is only called from my block test for right now, so I will remove it from here and move it into pocketnet_block_tests.cpp.

tawmaz committed:
Remove in-progress SQL database transaction validation functions

 - Remove GetTransactionHashes, verifydb, and CleanExpiredTransactions until those functions can be fully developed.
